src/base-traits.jl

```diff
@@ -1,15 +1,17 @@
 module BaseTraits
 using SimpleTraits
 
-export IsLeafType, IsBits, IsImmutable, IsContiguous, IsFastLinearIndex,
+using Compat
+
+export IsLeafType, IsBits, IsImmutable, IsContiguous, IsIndexLinear,
        IsAnything, IsNothing, IsCallable
 
 "Trait which contains all types"
 @traitdef IsAnything{X}
 @traitimpl IsAnything{X} <- (x->true)(X)
 
 "Trait which contains no types"
-typealias IsNothing{X} Not{IsAnything{X}}
+@compat IsNothing{X} = Not{IsAnything{X}}
```

Comment:
That NEWS.md entry is a bit confusing.  Anyway, it does seem to make a difference (unless something changed on 0.6 in the last 5 days):
```julia
julia> V = Int
Int64

julia> f() = convert(V,5.0)
f (generic function with 1 method)

julia> @code_warntype f()
Variables:
  #self#::#f

Body:
  begin
      return (Main.convert)(Main.V, 5.0)::Any
  end::Any

julia> const VV = Int
Int64

julia> ff() = convert(VV,5.0)
ff (generic function with 1 method)

julia> @code_warntype ff()
Variables:
  #self#::#ff

Body:
  begin
      return $(Expr(:invoke, MethodInstance for convert(::Type{Int64}, ::Float64), :(Main.convert), :(Main.VV), 5.0))
  end::Int64
```
So, probably better add the `const`.